tools/preprocess_pragmas.c

```diff
@@ -0,0 +1,146 @@
+
+// SPDX-FileCopyrightText: © 2024 ZeldaRET
+// SPDX-License-Identifier: CC0-1.0
+
+// Usage: preprocess_pragmas OOT_VERSION filename < source.c
+// The filename argument is only used for linemarkers.
+// Preprocess C source on stdin, writes to stdout
+// Replace `#pragma increment_block_number` with fake structs for controlling BSS ordering.
+// The names of these fake structs are expected to be increment_block_number_%d_%d with the first number indicating
+// the line number of the #pragma in the original source file. (this is for use by fix_bss.py)
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+const char str_pragma_increment_block_number[] = "#pragma increment_block_number";
+
+int main(int argc, char** argv) {
+#if 0
+    // for debugging with a specific file
+#undef stdin
+    FILE* stdin = fopen("src/code/fault.c", "r");
+    assert(stdin != NULL);
+#endif
```

Comment:
does this add anything over `< src/code/fault.c`?

Reply:
https://stackoverflow.com/a/455588 apparently `< input` is a feature of gdb

Reply:
thanks